contrib_bots/lib/xkcd.py

```diff
@@ -0,0 +1,134 @@
+from random import randint
+import logging
+import re
+import requests
+
+XKCD_TEMPLATE_URL = 'http://xkcd.com/%s/info.0.json'
+LATEST_XKCD_URL = 'http://xkcd.com/info.0.json'
+
+
+class XkcdHandler(object):
+    '''
+    This plugin provides several command that can be used for fetch comics from
+    xkcd. The bot looks for messages starting with "@xkcd" and responds with a
+    message with the comic based on provided commands.
+    '''
+
+    def usage(self):
+        return '''
+            This plugin allows users to fetch comics provided by xkcd.
+            Users should preface the command with "@xkcd".
+
+            There are several commands to use this bot:
+            - @xkcd help -> To show all commands the bot supports.
+            - @xkcd latest -> To fetch latest comic from xkcd.
+            - @xkcd random -> To fetch random comic from xkcd.
+            - @xkcd <comic_id> -> To fetch comic based on `<comic_id>`, e.g
+              `@xkcd 1234`.
+            '''
+
+    def triage_message(self, message):
+        # Return True if we want to (possibly) response to this message
+        original_content = message['content']
+        is_xkcd_called_pattern = re.compile(r'^(@xkcd$|@xkcd(\s+))')
+        is_xkcd_called = bool(is_xkcd_called_pattern.match(original_content))
+
+        return is_xkcd_called
+
+    def handle_message(self, message, client, state_handler):
+        xkcd_bot_response = get_xkcd_bot_response(message)
+
+        client.send_message(dict(
+            type='stream',
+            to=message['display_recipient'],
+            subject=message['subject'],
+            content=xkcd_bot_response,
+        ))
+
+
+class XkcdBotCommand(object):
+    LATEST = 0
+    RANDOM = 1
+    COMIC_ID = 2
+
+
+class XkcdNotFoundException(Exception):
```

Comment:
The convention is to use 'Error' as suffix, see https://www.python.org/dev/peps/pep-0008/#exception-names.